grommet/grommet
# Fixed handling of multi-level property names that are used as RangeSelectors in filters

Fixed handling of multi-level property names that are used as RangeSelectors in filters

Note that the badge doesn't go away when the filter is opened again and the range set back to its original value. It is possible to fix this, and I have a working fix, but it is a little complicated and inefficient so far. The user can still hit `Clear filters` to clear the badge

#### What does this PR do?

#### Where should the reviewer start?
DataForm.js

#### What testing has been done on this PR?
Manual via Storybook

#### How should this be manually tested?
In the `Data/Complex` story in storybook change the GDP range in the filter dialog to some other value, hit `Apply Filters` and observe that the badge on the filter toolbar icon appears.

#### Do Jest tests follow these best practices?


#### Any background context you want to provide?

#### What are the relevant issues?

Closes #7037 

#### Screenshots (if appropriate)
![image](https://github.com/grommet/grommet/assets/12054362/4bd34c8f-8343-46cd-b6e7-1a14be9a38d7)
![image](https://github.com/grommet/grommet/assets/12054362/f5992287-86e6-4994-a1a7-ed145a552e74)

#### Do the grommet docs need to be updated?

#### Should this PR be mentioned in the release notes?

#### Is this change backwards compatible or is it a breaking change?
